src/sqlfluff/core/rules/base.py

```diff
@@ -258,51 +258,55 @@ def create_after(
         """Create edit segments after the supplied anchor segment."""
         return cls("create_after", anchor_segment, edit_segments, source)
 
-    def has_template_conflicts(self, templated_file: TemplatedFile) -> bool:
-        """Does this fix conflict with (i.e. touch) templated code?"""
+    def get_fix_slices(
+        self, templated_file: TemplatedFile, within_only: bool
+    ) -> Set[RawFileSlice]:
+        """Returns slices touched by the fix."""
         # Goal: Find the raw slices touched by the fix. Two cases, based on
         # edit type:
-        # 1. "delete", "replace": Raw slices touching the anchor segment. If
-        #    ANY are templated, discard the fix.
+        # 1. "delete", "replace": Raw slices touching the anchor segment.
         # 2. "create_before", "create_after": Raw slices encompassing the two
         #    character positions surrounding the insertion point (**NOT** the
         #    whole anchor segment, because we're not *touching* the anchor
-        #    segment, we're inserting **RELATIVE** to it. If ALL are templated,
-        #    discard the fix.
+        #    segment, we're inserting **RELATIVE** to it.
         assert self.anchor.pos_marker
         anchor_slice = self.anchor.pos_marker.templated_slice
         templated_slices = [anchor_slice]
-        check_fn = any
 
+        # If "within_only" is set for a "create_*" fix, the slice should only
+        # include the area of code "within" the area of insertion, not the other
+        # side.
+        adjust_boundary = 1 if not within_only else 0
         if self.edit_type == "create_before":
             # Consider the first position of the anchor segment and the
             # position just before it.
             templated_slices = [
-                slice(anchor_slice.start, anchor_slice.start + 1),
-                slice(anchor_slice.start - 1, anchor_slice.start),
+                slice(anchor_slice.start - 1, anchor_slice.start + adjust_boundary),
```

Comment:
I made a small refactoring here that doesn't affect correctness, but is a bit more efficient. The old code created an array of two adjacent slices. The new code creates a single slice covering the same range.